src/main/java/net/fabricmc/tinyremapper/ClassInstance.java

```diff
@@ -148,10 +163,48 @@ public int getMrjVersion() {
 		return mrjVersion;
 	}
 
+	@Override
 	public String getSuperName() {
 		return superName;
 	}
 
+	@Override
+	public String getSignature() {
+		return signature;
+	}
+
+	@Override
+	public MemberHeader getField(String name, String desc) {
+		MemberInstance instance = this.resolvedMembers.get(MemberInstance.getFieldId(name, desc, desc == null));
+
+		if (instance != null) {
+			return new MemberHeader(this, instance.access, instance.name, instance.desc);
+		}
+
+		return null;
+	}
+
+	@Override
+	public MemberHeader getMethod(String name, String desc) {
+		MemberInstance instance = this.resolvedMembers.get(MemberInstance.getMethodId(name, desc));
+
+		if (instance != null) {
+			return new MemberHeader(this, instance.access, instance.name, instance.desc);
+		}
+
+		return null;
+	}
+
+	@Override
+	public Iterable<MemberHeader> allMembers() {
+		return MappedIterator.map(this.resolvedMembers.values(), i -> new MemberHeader(this, i.access, i.name, i.desc));
+	}
```

Comment:
`resolvedMembers` is a cache. It may or may or contains the member from parents' classes, which introduces inconsistency. I think a better way is needed.